libs/execution/src/lib/types/io-types/filesystem-node-directory.ts

```diff
@@ -0,0 +1,66 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+import { FileSystemNode } from './filesystem-node';
+import { FileSystemFile } from './filesystem-node-file';
+
+export class FileSystemDirectory extends FileSystemNode {
+  private children: FileSystemNode[] = [];
+  constructor(public override name: string) {
+    super(name);
+  }
+
+  override getNode(path: string): FileSystemNode | null {
+    const [firstPart, ...rest] = path.split('/');
+    if (firstPart !== this.name) {
+      return null;
+    }
+    if (rest.length === 0) {
+      return this;
+    }
+    for (const child of this.children) {
+      const f = child.getNode(rest.join('/'));
+      if (f) {
+        return f;
+      }
+    }
+    return null;
+  }
+
+  override putNode(path: string, node: FileSystemNode): FileSystemNode | null {
```

Comment:
splitted a little bit. But would prefer, having all necessary information on one view, since it is a recursive function. Added additionally comments to make is easier to read.